File: backend/app/schemas/instructions.py
from datetime import datetime

from pydantic import BaseModel, Field


class FlatInstructionWrite(BaseModel):
    id: str | None = None
    title: str = Field(min_length=1, max_length=255)
    video_url: str | None = Field(default=None, max_length=1000)
    video_name: str | None = Field(default=None, max_length=255)
    video_data: str | None = None
    description: str = Field(min_length=1)
    position: int = 0


class FlatInstructionSave(BaseModel):
    items: list[FlatInstructionWrite] = Field(default_factory=list)


class FlatInstructionRead(BaseModel):
    id: str
    title: str
    video_url: str | None = None
    video_name: str | None = None
    video_data: str | None = None
    description: str
    position: int
    building_id: str
    condominio_id: str
    created_at: datetime
    updated_at: datetime

File: frontend/src/services/instructions.ts
import { api } from "./api";

export type FlatInstruction = {
  id: string;
  title: string;
  video_url: string | null;
  video_name: string | null;
  video_data: string | null;
  description: string;
  position: number;
  building_id: string;
  condominio_id: string;
  created_at: string;
  updated_at: string;
};

export type FlatInstructionList = {
  flat: string;
  building_id: string;
  building_name: string;
  data: FlatInstruction[];
  count: number;
};

export const instructionsService = {
  async publicFlat(flat: string) {
    const { data } = await api.get<FlatInstructionList>(`/public-instructions/${flat}`);
    return data;
  },
  async flat(flat: string) {
    const { data } = await api.get<FlatInstructionList>(`/flat-instructions/${flat}`);
    return data;
  },
  async saveFlat(flat: string, items: Array<{ id?: string; title: string; video_url?: string | null; video_name?: string | null; video_data?: string | null; description: string; position: number }>) {
    const { data } = await api.put<FlatInstructionList>(`/flat-instructions/${flat}`, { items });
    return data;
  }
};
